packages/lit-html/src/lit-html.ts

```diff
@@ -11,6 +11,198 @@ const DEV_MODE = true;
+    }
+
+    export type CommitPartEntry =
+      | CommitNothingToChildEntry
+      | CommitText
+      | CommitNode
+      | CommitAttribute
+      | CommitProperty
+      | CommitBooleanAttribute
+      | CommitEventListener
+      | CommitToElementBinding;
+
+    export interface CommitNothingToChildEntry {
+      kind: 'commit nothing to child';
+      start: ChildNode;
+      end: ChildNode | null;
+      parent: Disconnectable | undefined;
+      options: RenderOptions | undefined;
+    }
+
+    export interface CommitText {
+      kind: 'commit text';
+      node: Text;
+      value: unknown;
+      options: RenderOptions | undefined;
+    }
+
+    export interface CommitNode {
+      kind: 'commit node';
+      start: Node;
+      parent: Disconnectable | undefined;
+      value: Node;
+      options: RenderOptions | undefined;
+    }
+
+    export interface CommitAttribute {
+      kind: 'commit attribute';
+      element: Element;
+      name: string;
+      value: unknown;
+      options: RenderOptions | undefined;
+    }
+
+    export interface CommitProperty {
+      kind: 'commit property';
+      element: Element;
+      name: string;
+      value: unknown;
+      options: RenderOptions | undefined;
+    }
+
+    export interface CommitBooleanAttribute {
+      kind: 'commit boolean attribute';
+      element: Element;
+      name: string;
+      value: boolean;
+      options: RenderOptions | undefined;
+    }
+
+    export interface CommitEventListener {
+      kind: 'commit event listener';
+      element: Element;
+      name: string;
+      value: unknown;
+      oldListener: unknown;
+      options: RenderOptions | undefined;
+      // True if we're removing the old event listener (e.g. because settings changed, or value is nothing)
+      removeListener: boolean;
+      // True if we're adding a new event listener (e.g. because first render, or settings changed)
+      addListener: boolean;
+    }
+
+    export interface CommitToElementBinding {
+      kind: 'commit to element binding';
+      element: Element;
+      value: unknown;
+      options: RenderOptions | undefined;
+    }
+  }
+}
+
+interface DebugLoggingWindow {
+  // Even in dev mode, we generally don't want to emit these events, as that's
+  // another level of cost, so only emit them when DEV_MODE is true _and_ when
+  // window.emitLitDebugEvents is true.
+  emitLitDebugLogEvents?: boolean;
+}
+
+/**
+ * Useful for visualizing and logging insights into what the Lit template system is doing.
+ *
+ * Compiled out of prod mode builds.
+ */
+const debugLogEvent = DEV_MODE
+  ? (event: LitUnstable.DebugLog.Entry) => {
+      const shouldEmit = (window as unknown as DebugLoggingWindow)
+        .emitLitDebugLogEvents;
+      if (shouldEmit) {
+        window.dispatchEvent(
+          new CustomEvent<LitUnstable.DebugLog.Entry>('lit-debug', {
+            detail: event,
+          })
+        );
+      }
+    }
+  : undefined;
+let debugLogId = 0;
```

Comment:
Will this variable be removed by Terser and Closure when not in DEV_MODE?

Reply:
Yep!